services/provider/server/server.go

```diff
@@ -106,12 +106,13 @@ func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.Onboard
 		return nil, status.Errorf(codes.Internal, "failed to get public key to validate onboarding ticket for consumer %q. %v", req.ConsumerName, err)
 	}
 
-	if err := validateTicket(req.OnboardingTicket, pubKey); err != nil {
+	onboardingTicket, err := decodeAndValidateTicket(req.OnboardingTicket, pubKey)
+	if err != nil {
 		klog.Errorf("failed to validate onboarding ticket for consumer %q. %v", req.ConsumerName, err)
 		return nil, status.Errorf(codes.InvalidArgument, "onboarding ticket is not valid. %v", err)
 	}
 
-	storageConsumerUUID, err := s.consumerManager.Create(ctx, req)
+	storageConsumerUUID, err := s.consumerManager.Create(ctx, req, int(onboardingTicket.StorageQuotaInGiB))
```

Comment:
Why do we need that cast to int? Is the type of value on the ticket different than the type of the value on the consumer?

Reply:
In `OnboardingTicket` we have it as uint and in storageconsumer spec we have it as int